### PR TITLE
Javascript 2 CA temp PR

### PR DESCRIPTION
**Pull request by lhvk**
Please review my PR and let me know what to better before my final delivery.

I am aware that the search function is not implemented yet and update post need some tweaks to function properly.

Thanks

### CODE REVIEW

**Comment by Fermain** [comment]
Hey Lars, I have listed all the issues I found (including the ones you know about already) in the Issues tab:

#8 
#7 
#6 
#5 

UX is not graded in this assignment so I would prioritise the bugs. I have suggested labels for each, you can create these yourself if they are not listed.

You are not far away from a pass but more time is still needed on this.

**Comment by lhvk** [comment]
Every issue should now be solved.

File: js/api/getAllPosts.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * @param {getAllPosts} fetchOptions options for the fetch
+ */
+/*======================================================================================================
+Get all posts
+======================================================================================================*/
+
+export const getAllPosts = async function (url, fetchOptions) {
+  try {
+    //
+    fetchOptions;
```
Comment (Fermain) [inline js/api/getAllPosts.js]: Can you explain this line?

File: js/api/getSinglePost.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"use strict";
+
+/**
+ *
+ * @param {*} url url for the fetch
```
Comment (Fermain) [inline js/api/getSinglePost.js]:
```suggestion
 * @param {string} url url for the fetch
```

File: js/api/getSinglePost.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * @param {*} fetchOptions options for the fetch
```
Comment (Fermain) [inline js/api/getSinglePost.js]:
```suggestion
 * @param {Object} fetchOptions options for the fetch
```

File: js/api/getSinglePost.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * @param {*} getHtml html for the UI of the fetch
```
Comment (Fermain) [inline js/api/getSinglePost.js]:
```suggestion
 * @param {function} getHtml html for the UI of the fetch
```

File: js/api/newPost.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+"use strict";
+
+/**
+ * @param {addNewPost} url for the fetch request
```
Comment (Fermain) [inline js/api/newPost.js]:
```suggestion
 * @param {string} url for the fetch request
```

File: js/api/newPost.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ * @param {addNewPost} newPostData data for the new post from the form input
```
Comment (Fermain) [inline js/api/newPost.js]:
```suggestion
 * @param {Object} newPostData data for the new post from the form input
```

File: js/components/components.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+======================================================================================================*/
+// Change document title
+export const changeDocTitle = (title) => {
+  document.title = `Social Media Page | ${title}`;
+};
+
+// Sign out eventhandler
+export const signOut = () => {
+  document.querySelector("#signOut").addEventListener("click", () => {
+    localStorage.clear();
```
Comment (Fermain) [inline js/components/components.js]:
```suggestion
    localStorage.remove("token");
```

Clear is too powerful, use remove with a key instead. You may want to keep profile data in storage to make the login process easier

**Review by Fermain — CHANGES_REQUESTED** [review]
There are some minor changes to be made and some suggestions. The most important thing is double checking your JSdoc params. These should not use `*` and the type should relate to the data type of the argument itself.

The main code improvement is one that you won't be able to make now as it relates to structure. I would like to see more `encapsulation` in future instead of having many different scripts for different pages, it would be good to see a single entry point and some sort of routing strategy.

File: js/components/components.js (hunk 7 above)
Comment (lhvk) [reply js/components/components.js]: Remove does not work. removeItem redirects me to the login, but the local storage is not cleared. I can still access the feed if I write it directly in the url.

File: js/api/getAllPosts.js (hunk 1 above)
Comment (lhvk) [reply js/api/getAllPosts.js]: I removed the const for the fetchOptions and I passed it in its place by mistake. I see that i have it in the response which is what matters. I have now deleted this line.

File: js/components/components.js (hunk 7 above)
Comment (lhvk) [reply js/components/components.js]: Added removeItem with the correct key and it now clears from local storage upon signout.

**Review by Fermain — APPROVED** [review]

**Review by Fermain — APPROVED** [review]
Well done

**Pull request merged**